src/install/dracut-install.c

```diff
@@ -1515,7 +1515,114 @@ static bool check_module_path(const char *path)
         return true;
 }
 
-static int install_dependent_module(struct kmod_module *mod, int *err)
+static int find_module_from_sysfs_node(struct kmod_ctx *ctx, const char *sysfs_node, int sysfs_node_len,
+                                       struct kmod_list **modules)
+{
+        char modalias_path[PATH_MAX] = {0};
+        snprintf(modalias_path, sizeof(modalias_path), "%.*s/modalias", sysfs_node_len, sysfs_node);
+
+        _cleanup_close_ int modalias_file = -1;
+        if ((modalias_file = open(modalias_path, O_RDONLY | O_CLOEXEC)) == -1)
+                return 0;
+
+        char alias[getpagesize()];
```

Comment:
any reason why not just `PAGE_SIZE`?

Reply:
AFAICT `PAGE_SIZE` is only declared in the kernel headers. I came across https://stackoverflow.com/questions/37897645/page-size-undeclared-c that suggested to use `getpagesize()` instead 

But I see there's also a `page_size()` function defined in `src/install/util.c` that calls `sysconf(_SC_PAGESIZE)` Maybe I should use that instead, what do you think?

Reply:
of course; POSIX requires `PAGESIZE`, glibc doesn't provide it. yeah, use that then.